### DISCUSSION TITLE
how to handle 'public' folder and add favicon in React App?

### DISCUSSION THREAD

**Discussion by kongmengfei** [webpack/webpack#16506] [Q&A]
Hello All,

I want to create a project (react + typescript +webpack) from zero. Firstly I copy the folder structure from the CRA template and then write my own webpack config file. below is my project: 

- https://github.com/kongmengfei/demo01

Folders View: 

![image](https://user-images.githubusercontent.com/57654821/203508126-3a052460-a63a-413f-8c2b-0b85791b33da.png)

But how to handle the public folder? now the index.html still shows an invalid favicon? it seems that the public folder did not get any compilation. 

another question is that the final bundle js is over 2MB. this is too large. 

![image](https://user-images.githubusercontent.com/57654821/203509366-07d7fd14-52c7-48cb-8d12-64f628d60468.png)

So I try to create several subfolders in src and then compile them into several js files, is that ok?

appreciate any advice!

**Comment by alexander-akait** [comment]
> another question is that the final bundle js is over 2MB. this is too large.

Do you build for production (i.e. you should use `terser-webpavck-plugin`)

> But how to handle the public folder? now the index.html still shows an invalid favicon? it seems that the public folder did not get any compilation.

Do you have plugins? How you generate favicons?
